plugins/postgres/dbt/adapters/postgres/impl.py

```diff
@@ -131,3 +131,13 @@ def valid_incremental_strategies(self):
         Not used to validate custom strategies defined by end users.
         """
         return ["append", "delete+insert"]
+
+    @property
+    def constraint_support(self):
+        return {
+            "not_null": ConstraintSupport.ENFORCED,
```

Comment:
we could use the [`ConstraintType` enum](https://github.com/dbt-labs/dbt-core/blob/main/core/dbt/contracts/graph/nodes.py#L143) for keys in this dictionary instead of strings.